src/blocks/rsvp-response-v2/rsvp-manager.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * WordPress dependencies.
+ */
+import { useState } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+import { FormTokenField, SelectControl } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies.
+ */
+import { getFromGlobal, setToGlobal } from '../../helpers/globals';
+
+/**
+ * Component for displaying and managing RSVP responses.
+ *
+ * This component renders a user interface for managing RSVP responses to an event.
+ * It includes options for attending, being on the waiting list, or not attending,
+ * and updates the status based on user interactions. The component also listens for
+ * changes in RSVP status and updates the state accordingly.
+ *
+ * @param {Object}   root0                  The destructured props object.
+ * @param {string}   root0.defaultStatus    The current default status for the RSVP response.
+ * @param {Function} root0.setDefaultStatus The function to update the defaultStatus state.
+ * @since 1.0.0
+ *
+ * @return {JSX.Element} The rendered RSVP response component.
+ */
+const RsvpManager = ({ defaultStatus, setDefaultStatus }) => {
+	const responses = getFromGlobal('eventDetails.responses');
+	const postId = getFromGlobal('eventDetails.postId');
+	const [rsvpResponse, setRsvpResponse] = useState(responses);
+	const attendees = rsvpResponse[defaultStatus].responses;
+
+	/**
+	 * Fetches user records from the core store via getEntityRecords.
+	 * Returns userList containing the list of user records.
+	 */
+	const { userList } = useSelect((select) => {
+		const { getEntityRecords } = select(coreStore);
+
+		const users = getEntityRecords('root', 'user', {
+			per_page: -1,
```

Comment:
@mauteri  Can we use something other than `-1` here to avoid potential performance issues?